libtenzir/src/os.cpp

```diff
@@ -413,7 +452,7 @@ auto socket_state_to_string(auto proto, auto state) -> std::string_view {
 
 auto darwin_os::fetch_sockets() -> std::vector<socket> {
   auto result = std::vector<socket>{};
-  for (const auto& proc : fetch_processes()) {
+  for (const auto& proc : fetch_processes(std::nullopt)) {
```

Comment:
```suggestion
  for (const auto& proc : fetch_processes()) {
```

Reply:
If you make `= {}` a default arg, the API looks a bit nicer, as `nullopt` as arg is not very semantic.